src/main/java/org/jabref/gui/preferences/appearance/AppearanceTab.java

```diff
@@ -27,6 +31,16 @@
 
     private final ControlsFxVisualizer validationVisualizer = new ControlsFxVisualizer();
 
+    // The fontSizeFormatter formats the input given to the fontSize spinner so that non valid values cannot be entered.
+    private TextFormatter<Integer> fontSizeFormatter = new TextFormatter<Integer>(new IntegerStringConverter(), 9,
```

Comment:
As this might be handy also in other places, I would propose to extract this to a static method in the a helper class `gui.util.TextFormatter`